.github/workflows/maven.yml

```diff
@@ -0,0 +1,37 @@
+name: maven
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, labeled, unlabeled, synchronize]
+
+permissions:
+  contents: read
+
+env:
+  GRADLE_OPTS: "-Dorg.gradle.jvmargs=-Xmx4g -Dorg.gradle.daemon=false -Dkotlin.incremental=false"
+
+jobs:
+  test_maven:
```

Comment:
Would you consider adding this to the existing `build.yml` ? I don’t think Maven is different-enough from the other checks in that file, and it’s fewer YML files to keep up-to-date for things like `on` and `env`